report-jtreg-comparator/src/main/java/io/jenkins/plugins/report/jtreg/main/comparator/StackTraceCompare.java

```diff
@@ -154,8 +161,26 @@ private static List<JtregPluginServicesLinkWithTooltip> createTooltip(String res
             String jobId2 = Builds.getBuildNumber(new File(jobBuilds.get(x)));
             list.add(new JtregPluginServicesLinkWithTooltip(" * show diff against " + (x + 1), getDiffLink(buildName, buildId, buildName2, jobId2, test, diffUrl), null));
         }
-        list.add(new JtregPluginServicesLinkWithTooltip(" * use this as base (not yet working properly)",
-                "http://copy.this/--set-referential " + buildName + ":" + buildId + "#" + id.replaceAll("#", "%23"),
+
+        List<String> mutableArg = new ArrayList<>(VariantComparator.copyOfArgs);
+        for (int x = 0; x < mutableArg.size(); x++) {
+            mutableArg.set(x,  mutableArg.get(x).replaceAll("#", "%23"));
+            if (mutableArg.get(x).equals(ComparatorArgDeclaration.setReferentialArg.getName())) {
+                mutableArg.remove(x);
+                //Fragile but we expected parser did job correctly
+                mutableArg.remove(x);
+                x = x - 1;
+            }
+
+        }
+        mutableArg.add(ComparatorArgDeclaration.setReferentialArg.getName());
+        mutableArg.add(buildName + ":" + buildId);
+
+        String strArgs = String.join(" ", mutableArg);
+        strArgs = strArgs.replace(' ', '+');
```

Comment:
Please add tests for those three logic parts. That means, you have to split it to three method - one which removes  all redundant  `setReferentialArg` , second which is oadding the only one referential tag, and third, which is sanitizing that. Each emthod shoudl have its own tests. Simialrly the conenction should have some tests